katdal/visdatav4.py

```diff
@@ -301,6 +347,26 @@ def _before(date):
         _add_sensor_alias(self.sensor, 'Antennas/array/activity', 'obs_activity')
         _add_sensor_alias(self.sensor, 'Antennas/array/target', 'cbf_target')
 
+        # ------ Extract correlator settings ------
+
+        self.sensor['Correlator/instrument'] = CategoricalData(
+            [self._cbf_instrument], all_dumps)
```

Comment:
We shouldn't assume that all the relevant CBF streams come from the same instrument: this is the instrument for the baseline-correlation-products stream.